### ISSUE TITLE
Tracking pipeline memory usage and relation size via API and UI

### ISSUE THREAD

**Issue by lalithsuresh** [feldera/feldera#1343]
With long running pipelines, it's currently hard to tell why memory usage is going up. It'd be good to be able to tell how big each relation is (both tables and views), and surface this info through the APIs and the UI.

@mihaibudiu mentions that we can incrementally track sizes by inserting count(*) queries from the compiler, or track this information in system tables (which we don't yet have). I suggest we start with the former for now and switch to systems tables when we have support for them.

**Comment by ryzhyk** [comment]
The way to do this now would be to add the count stream along with neighborhood and sample streams to output handles in the catalog.

**Comment by mihaibudiu** [comment]
I am suggesting a *single* view with information. It is rather tedious to browse a view for a single row.

**Comment by lalithsuresh** [comment]
We could generate a single view like this for relation sizes.

```
create view system__relation_sizes as 
(select "t1", count(* ) from t1)
union
(select "t2", count(*) from t2) .....
```

**Comment by mihaibudiu** [comment]
Should I put this in the next milestone?
It should be controlled by a compiler flag.

**Comment by lalithsuresh** [comment]
Yes, go ahead and add it to the milestone please.

**Comment by ryzhyk** [comment]
I don't think this is the right way to implement this. Adding extra views will introduce a lot of extra circuitry and slow down compilation. Plus there's an extra config option that needs to be somehow propagated through the API and UI.

A better solution would be to have the catalog to always insert the count operator, and then we can expose it through API and UI, like we do with neighborhoods.

**Comment by mihaibudiu** [comment]
I thought we were moving more code into the compiler from the runtime, not the other way around.

**Comment by ryzhyk** [comment]
The catalog logic should move into the compiler, yes.

**Comment by mihaibudiu** [comment]
So what do I do for the next milestone?

**Comment by ryzhyk** [comment]
For this issue, probably nothing.

**Comment by mihaibudiu** [comment]
I think that the profiler API and UI solve this issue.
